x-pack/plugins/visualization_lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { render } from 'react-dom';
+import { Datasource, Operation, DataType } from '../';
+
+interface IndexPatternPrivateState {
+  query: object;
+}
+
+class IndexPatternDatasource implements Datasource<IndexPatternPrivateState> {
+  private state: IndexPatternPrivateState;
+
+  constructor(state?: IndexPatternPrivateState) {
+    if (state) {
+      this.state = state;
+    } else {
+      this.state = {
+        query: {},
+      };
+    }
+  }
+
+  toExpression() {
+    return `${JSON.stringify(this.state.query)}`;
+  }
+
+  renderDataPanel({ domElement }: { domElement: Element }) {
+    render(<div>Index Pattern Data Source</div>, domElement);
+  }
+
+  getPublicAPI() {
+    return {
+      getTableSpec: () => [],
+      getOperationForColumnId: () => ({
+        id: '',
+        // User-facing label for the operation
+        label: '',
+        dataType: 'string' as DataType,
+        // A bucketed operation has many values the same
+        isBucketed: false,
+      }),
+
+      // Called by dimension
+      getDimensionPanelComponent: (props: any) => (
+        domElement: Element,
+        operations: Operation[]
+      ) => {},
+
+      removeColumnInTableSpec: (columnId: string) => [],
+      moveColumnTo: (columnId: string, targetIndex: number) => {},
+      duplicateColumn: (columnId: string) => [],
+    };
+  }
+
+  getDatasourceSuggestionsForField() {
+    return [];
+  }
+
+  getDatasourceSuggestionsFromCurrentState() {
```

Comment:
I already forget what scenario this one is for. A comment would probably help. Also I think this and the previous method could just be: `getSuggestionsForField` and `getSuggestionsFromCurrentState` or whatever. (e.g. drop the Datasource part of the name). Lastly, I think we probably should have suggestions for fields (plural) instead of singular, but that's mainly because I like the multi-select interaction of some of the products we reviewed.

Reply:
Will update